mentoring/mcq.py

```diff
@@ -0,0 +1,196 @@
+                return self._(u"Correct")
+            return self._(u"Acceptable")
+        else:
+            if len(self.correct_choices) == 1:
+                return self._(u"Wrong")
+            return self._(u"Not Acceptable")
+
+    def submit(self, submission):
+        log.debug(u'Received MCQ submission: "%s"', submission)
+
+        correct = submission in self.correct_choices
+        tips_html = []
+        for tip in self.get_tips():
+            if submission in tip.values:
+                tips_html.append(tip.render('mentoring_view').content)
+
+        if tips_html:
+            formatted_tips = ResourceLoader(__name__).render_template('templates/html/tip_choice_group.html', {
+                'tips_html': tips_html,
+            })
+
+        self.student_choice = submission
+        result = {
+            'submission': submission,
+            'status': 'correct' if correct else 'incorrect',
+            'tips': formatted_tips if tips_html else None,
+            'weight': self.weight,
+            'score': 1 if correct else 0,
+        }
+        log.debug(u'MCQ submission result: %s', result)
+        return result
+
+    def author_edit_view(self, context):
+        """
+        The options for the 1-5 values of the Likert scale aren't child blocks but we want to
+        show them in the author edit view, for clarity.
+        """
+        fragment = Fragment(u"<p>{}</p>".format(self.question))
+        self.render_children(context, fragment, can_reorder=True, can_add=False)
+        fragment.add_content(loader.render_template('templates/html/questionnaire_add_buttons.html', {}))
+        fragment.add_css_url(self.runtime.local_resource_url(self, 'public/css/questionnaire-edit.css'))
+        return fragment
+
+    def validate_field_data(self, validation, data):
+        """
+        Validate this block's field data.
+        """
+        super(MCQBlock, self).validate_field_data(validation, data)
+
+        def add_error(msg):
+            validation.add(ValidationMessage(ValidationMessage.ERROR, msg))
+
+        def choice_name(choice_value):
+            for choice in self.human_readable_choices:
+                if choice["value"] == choice_value:
+                    return choice["display_name"]
+            return choice_value
+
+        all_values = set(self.all_choice_values)
+        correct = set(data.correct_choices)
+
+        if not all_values:
+            add_error(self._(u"No choices set yet."))
+        elif not correct:
+            add_error(
+                self._(u"You must indicate the correct answer[s], or the student will always get this question wrong.")
+            )
+        if len(correct) < len(data.correct_choices):
+            add_error(self._(u"Duplicate correct choices set"))
+        for val in (correct - all_values):
+            add_error(
+                self._(u"A choice value listed as correct does not exist: {choice}").format(choice=choice_name(val))
```

Comment:
@nedbat I think that it is already like that? It just looks like it's wrong if you focus on the end parens because of the extra `choice_name()` wrapper around `val`